### PR TITLE
Add VZ prepared-host evidence tracker

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary
- Add a prepared-host `vz_linux` evidence tracker with packet fields, expected skips, acceptance checklist, residual gaps, and recording guidance.
- Link the tracker from the host-gated CI policy, macOS operator notes, and sandbox roadmap.
- Add focused infrastructure tests that guard the evidence contract and manual/host-gated execution boundary.

## Verification
- `/Users/macbook-dev/Documents/GitHub/tldw_server2/.venv/bin/python -m pytest tldw_Server_API/tests/Infrastructure/test_vz_linux_host_gated_workflow.py -q`
- `git diff --check origin/dev..HEAD`

## Bandit
Skipped: docs/test-only change; no production Python/runtime code changed.

## Change summary
Human-authored change summary pending.

---
## Summary by cubic
Adds a prepared-host evidence tracker for Apple silicon `vz_linux` runs that defines the evidence packet, acceptance checklist, expected skips, residual gaps, and recording guidance. Cross-links it from the host-gated CI policy, macOS operator notes, and the sandbox roadmap, and adds hardened infra tests that enforce the evidence contract and keep real VM execution manual/host-gated; no runtime or workflow-trigger changes.

<sup>Written for commit 8296d0361529f9aad94742b135bf1b74f5832804. Summary will update on new commits. <a href="https://cubic.dev/pr/rmusser01/tldw_server/pull/1847?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: tldw_Server_API/tests/Infrastructure/test_vz_linux_host_gated_workflow.py [hunk 1]
```diff
@@ -175,3 +176,34 @@ def test_vz_linux_host_gated_acceptance_policy_doc_exists_and_references_workflo
...
+    assert "real `vz_linux` ephemeral execution" in tracker_lower  # nosec B101
+    assert "same-session vm reuse" in tracker_lower  # nosec B101
+    assert "helper build/signing" in tracker_lower  # nosec B101
+    assert "artifact" in tracker_lower  # nosec B101
+    assert "expected skips" in tracker_lower  # nosec B101
+    assert "residual gaps" in tracker_lower  # nosec B101
+    assert "do not paste secrets" in tracker_lower  # nosec B101
+
+
+def test_vz_linux_prepared_host_evidence_tracker_keeps_real_vm_runs_gated() -> None:
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/Infrastructure/test_vz_linux_host_gated_workflow.py:224]: Verified against current code: `test_vz_linux_prepared_host_evidence_tracker_keeps_real_vm_runs_gated() -> None` already declares the `-> None` return type at the current line. No code change needed for this thread.
